Final_project/sql/transfer_sales_from_bronze_to_silver.sql

```diff
@@ -0,0 +1,20 @@
+DELETE FROM `{{ params.project_id }}.2_silver.sales`
+WHERE DATE(purchase_date) = "{{ ds }}";
+
+INSERT  `{{ params.project_id}}.2_silver.sales` (
+    client_id,
+    purchase_date,
+    product_name,
+    price,
+    _id,
+    _logical_dt
+)
+SELECT
+    CAST(CustomerId AS INTEGER),
+    CAST(REPLACE(PurchaseDate, '/', '-') AS DATE),
+    Product,
+    CAST(REPLACE(REPLACE(Price, 'USD', ''), '$', '') AS INTEGER),
+    _id,
+    _logical_dt
+FROM `{{ params.project_id}}.1_bronze.sales`
+WHERE CAST(REPLACE(PurchaseDate, '/', '-') AS DATE) = "{{ ds }}";
```

Comment:
You partitioned the Bronze layer on `_logical_dt` field, but query here by `PurchaseDate`. It means that your query scans all the partitions and it costs a lot of money.